### PR TITLE
Work In Progress: Feature/authentication

### PR DESCRIPTION
**Pull request by APM246**
Having issues with creating a superuser to test out login page. The error is "create_superuser() missing 1 required positional argument: phone". The issue would be fixed if migrations were applied correctly but it doesn't seem to recognise my changes to create_superuser().

#22

### CODE REVIEW

File: app/admin_settings/migrations/0001_initial.py
Comment (zollf) [inline app/admin_settings/migrations/0001_initial.py]: why are these removed? This is a key, value table. These key-values are only every used once and there is no need to make them create-able from the admin panel

File: app/resources/templates/registration/login.html
Comment (zollf) [inline app/resources/templates/registration/login.html]: this file should be in templates, no need to segment templates into folders. 
This would also make this `{% extends 'base.html' %} `

File: app/authentication/models.py
Comment (zollf) [inline app/authentication/models.py]: `user = self.model(email=email, phone=phone,  **extra_fields)` u forgot the comma

File: app/authentication/models.py
Comment (zollf) [inline app/authentication/models.py]: leave this out for now

File: app/authentication/models.py
Comment (zollf) [inline app/authentication/models.py]: `default=False`

File: app/authentication/models.py
Comment (zollf) [inline app/authentication/models.py]: `null=True`

File: app/authentication/models.py
Comment (zollf) [inline app/authentication/models.py]: `default=0`

File: app/core/settings.py
Comment (zollf) [inline app/core/settings.py]: `admin/`

File: app/resources/templates/registration/login.html
Comment (zollf) [inline app/resources/templates/registration/login.html]: please use `<input type="text" name="username" id="username"/>` just for consistence sake, and same with the label

File: app/authentication/models.py
Comment (zollf) [inline app/authentication/models.py]: i dont think u need these checks, as youre setting them to True immediately before

**Review by zollf — CHANGES_REQUESTED** [review]

**Comment by zollf** [comment]
You can delete you db my running `docker-compose rm - v mysql` then you'll need to initialize again `docker-compose up mysql` then rerun the migrations. 

You can delete your migration file and redo them as we haven't committed them into `main` if you need to.

File: app/authentication/models.py
Comment (APM246) [reply app/authentication/models.py]: Do u mean leave out hub_id?

Comment (zollf) [reply app/authentication/models.py]: yea leave out hub_id, github just highlights chunk of code when i comment on a line. By the bottom line is the one i mean so hub_id

File: app/resources/templates/registration/login.html
Comment (APM246) [reply app/resources/templates/registration/login.html]: Doesn't this take advantage of Django's Form objects which improves syntax clarity/simplicity

**Comment by APM246** [comment]
> You can delete you db my running `docker-compose rm - v mysql` then you'll need to initialize again `docker-compose up mysql` then rerun the migrations. 
> You can delete your migration file and redo them as we haven't committed them into `main` if you need to.

Getting this error after deleting SQL image + migrations and re-running migrations from scratch: 
django.db.utils.OperationalError: (1050, "Table 'django_admin_log' already exists")

Trying to investigate

File: app/resources/templates/registration/login.html
Comment (zollf) [reply app/resources/templates/registration/login.html]: since youre using LoginView and its doing it all for you. Its fine

**Comment by zollf** [comment]
maybe try deleting again. I dont see why deleting the mysql volume wouldn't remove that specific table. Removing the volume removes the whole db. The only way i can see this happening is if you arent deleting the correct database. https://gyazo.com/db8083fc1876e11ef6c36102f0371319

**Comment by zollf** [comment]
i just noticed i didnt type the command correct *facepalm* its `docker-compose rm -v mysql` no space between - and v

**Comment by APM246** [comment]
Nvm ended up fixing it, ran some commands in the wrong order. Will try to commit new changes soon.

File: app/admin_settings/migrations/0001_initial.py
Comment (APM246) [reply app/admin_settings/migrations/0001_initial.py]: Hmm for some reason this code got removed when I ran migrations from scratch

Comment (APM246) [reply app/admin_settings/migrations/0001_initial.py]: Oh nvm was that code in a later commit? Cause I haven't merged from main recently

Comment (zollf) [reply app/admin_settings/migrations/0001_initial.py]: `git checkout origin/main app/admin_settings/migrations/0001_initial.py`

Comment (APM246) [reply app/admin_settings/migrations/0001_initial.py]: I believe insertData should have "self" as the first arg since it's a method

Comment (zollf) [reply app/admin_settings/migrations/0001_initial.py]: isnt it a static method?

anyway i dont know if it matters, it works for you right?

Comment (APM246) [reply app/admin_settings/migrations/0001_initial.py]: Oh right yea. In that case I've added @staticmethod to getKeys() and getDict() (to get rid of the VS Code squiggy line)

Comment (zollf) [reply app/admin_settings/migrations/0001_initial.py]: Thanks. What plugin is that, ill install it

**Comment by APM246** [comment]
Authentication now works but I'm not ready to merge (still have a few things to tweak)

**Comment by zollf** [comment]
Thanks. Ill add user page in the backend so we can hook it up easily straight after this gets merged. You also have conflicts btw.

**Comment by APM246** [comment]
Yep that's one of the things I wanted to tweak, merging main atm

**Comment by APM246** [comment]
The Django test needs to be modified because I've now added login_required and staff_member_required decorators to the admin view

**Comment by zollf** [comment]
ill do it now

**Comment by APM246** [comment]
Could I try doing it?

**Comment by zollf** [comment]
yea go for it

File: app/resources/templates/base.html
Comment (zollf) [inline app/resources/templates/base.html]: conflict lines

**Review by zollf — CHANGES_REQUESTED** [review]

File: app/resources/templates/base.html
Comment (APM246) [reply app/resources/templates/base.html]: Jeez I'm really out of it, I can't even get a merge right

**Review by zollf — APPROVED** [review]
Good work

**Pull request merged**
